izer/assets.py

```diff
@@ -112,3 +112,162 @@ def from_template(
             else:
                 shutil.copy(os.path.join(base, source, source_path),
                             os.path.join(target, test_path))
+
+
+def vscode(
+    out_stem: str,
+    out_branch: str,
+    part_num: str = "",
+    board: str = "",
+    elf_file: str = "",
+    defines: str = "",
+    i_paths: str = "",
+    v_paths: str = "",
+):
+    """
+    Generate vscode project files from a template, and places
+    the contents of the template folder `assets/vscode` in `out_stem/out_branch`.
+    Optional arguments will load from the global state unless overridden.
+
+    Parameters:
+        out_stem:  Output root directory.
+        out_branch:  Output sub-directory.
+        (optional) part_num:  Target part number.  Ex: MAX78000
+        (optional) board:  Target board, case sensitive.  Ex:  EvKit_V1, FTHR_RevA, etc.
+        (optional) elf_file:  Sets the name of the output file.  Ex:  hello_world.elf
+        (optional) i_paths:  Space-separated include paths the C/C++ parser should use.
+        (optional) defines:  Space-separated compiler definitions the C/C++ parser should use.
+        (optional) v_paths:  Space-separated additional browse paths the C/C++ parser should use.
+
+    Returns:
+        Nothing
+    """
+
+    template_dir = os.path.join("assets", "vscode")  # Where to find the VS Code template directory
+    template_prefix = "template"  # Filenames beginning with this will have substitution
+
+    # Load defaults from global state...
+    # ---
+    if part_num == "":
+        assert tc.dev is not None
+        part_num = tc.dev.partnum
+
+    if board == "":
+        board = state.board_name
+
+    if elf_file == "":
+        if state.riscv:
+            # RISC-V projects will look for ...-combined.elf
+            elf_file = "${config:proj_name}-combined.elf"
+        else:
+            # Default is project name (which defaults to folder name in template)
+            elf_file = "${config:proj_name}.elf"
+
+    if defines == "":
+        defines = state.defines
+
+    if i_paths == "":
+        i_paths = state.eclipse_includes  # TODO: rename state var to 'includes'?
+
+    if v_paths == "":
+        pass  # TODO: Support adding browse paths on command line
+    # ---
+
+    tmp = []  # Work-horse list, linter be nice
+    # Parse defines...
+    # ---
+    tmp = defines.split(" ")
+
+    if state.defines_arm != "":
+        # Split & append Arm defines
```

Comment:
Can't you just do tmp += state.defines_arm.split(" ")

Reply:
You're correct.  I was using `tmp.append(state.defines_arm.split(" "))` initially, but that was appending _as_ a list...  ie:  `tmp = [define1, define2, [define3, define4, define5]]`, etc.

Using `+=` instead of `tmp.append` appends the list elements instead of the list itself.  Switched the code over to use `+=` which cleans it up.